exercism/bracket-push/bracket-push.js

```diff
@@ -0,0 +1,40 @@
+function bracket(str){
+var paren = ['(', ')']
+var curly = ['{','}']
+var square = ['[',']']
+var closings = [')', '}', ']']
+
+var arr = str.split('') //['{','}']
+var openingBracketsArr = []
+
+  for( var i in arr ){
+    if( closings.includes( arr[i] ) ){
+    var bracketType = check( arr[i] )
+      if( check(openingBracketsArr.pop()) !== bracketType ){
+        return false
+      }
+    }else{
+      openingBracketsArr.push( arr[i] )
+    }
+  }
+
+  function check(element){
+    if( paren.includes(element) ){
+      return 'Im a paren'
+    }else if( curly.includes(element) ){
+      return 'Im a curly'
+    }else if( square.includes(element) ){
+      return 'Im a square'
+    }else{
```

Comment:
This is where using an object like I mentioned above could come in handy.  This function is helping you to use the closing brace to look up the opening brace.  If you had an object like `let brackets = { ')': '(' }`, then you could do this lookup directly.  `brackets[')']` would return `'('`.